File: apis/src/main/java/com/beat/apis/home/application/dto/HomePromotionDetail.java
package com.beat.apis.home.application.dto;

import com.beat.global.support.jackson.CdnImageUrl;

public record HomePromotionDetail(
	Long promotionId,
	@CdnImageUrl String promotionPhoto,
	Long performanceId,
	String redirectUrl,
	boolean isExternal,
	String carouselNumber
) {

	public static HomePromotionDetail of(Long promotionId, String promotionPhoto, Long performanceId,
		String redirectUrl,
		boolean isExternal, String carouselNumber) {
		return new HomePromotionDetail(promotionId, promotionPhoto, performanceId, redirectUrl, isExternal,
			carouselNumber);
	}
}

File: apis/src/main/kotlin/com/beat/apis/home/api/response/HomePromotionDetail.kt
package com.beat.apis.home.api.response

import com.beat.apis.home.application.result.HomePromotionResult
import com.beat.global.support.jackson.CdnImageUrl
import com.fasterxml.jackson.annotation.JsonProperty

@ConsistentCopyVisibility
data class HomePromotionDetail private constructor(
    val promotionId: Long?,
    @field:CdnImageUrl val promotionPhoto: String?,
    val performanceId: Long?,
    val redirectUrl: String?,
    @get:JsonProperty("isExternal")
    val isExternal: Boolean,
    val carouselNumber: String?,
) {
    companion object {
        fun from(result: HomePromotionResult): HomePromotionDetail = HomePromotionDetail(
            promotionId = result.promotionId,
            promotionPhoto = result.promotionPhoto,
            performanceId = result.performanceId,
            redirectUrl = result.redirectUrl,
            isExternal = result.isExternal,
            carouselNumber = result.carouselNumber,
        )
    }
}
